### PR TITLE
fix(provider): inventory adjustment for multiple replicas

### PR DESCRIPTION
**Pull request by troian**
add kube inventory tests
fixes #SUPPORT-108

Signed-off-by: Artur Troian <troian.ap@gmail.com>

### CODE REVIEW

File: provider/cluster/kube/inventory.go [hunk 1]
```diff
@@ -162,12 +162,10 @@ nodes:
 				currInventory.storageClasses = storageClasses
 			}
 
-			if res.Count > 0 {
-				currResources = append(currResources, res)
+			if resources[rIdx].Count == 0 {
```
Comment (boz) [inline provider/cluster/kube/inventory.go:165]: `> 0`?

**Comment by boz** [comment]
> What is motivating this change?

Providers aren't bidding if replica count is greater than 1 apparently.

**Comment by hydrogen18** [comment]
We're talking about the replica count on the SDL?

**Comment by boz** [comment]
> We're talking about the replica count on the SDL?

yeah

**Comment by hydrogen18** [comment]
I can deploy this on master no problem

```
---
version: "2.0"

services:
  web:
    image: quay.io/ovrclk/demo-app
    expose:
      - port: 80
        as: 80
        http_options:
          max_body_size: 2097152
          next_cases:
            - off
        accept:
          - hello.localhost
        to:
          - global: true
  bew:
    image: quay.io/ovrclk/demo-app
    expose:
      - port: 80
        as: 80
        accept:
          - hello1.localhost
        to:
          - global: true

profiles:
  compute:
    web:
      resources:
        cpu:
          units: 0.1
        memory:
          size: 16Mi
        storage:
          size: 128Mi
  placement:
    westcoast:
      attributes:
        region: us-west
      pricing:
        web: 
          denom: uakt
          amount: 1000

deployment:
  web:
    westcoast:
      profile: web
      count: 2
  bew:
    westcoast:
      profile: web
      count: 1
```

my pods in k8s look like this 
Screenshot from 2022-05-12 10-32-03

File: provider/cluster/kube/inventory.go (hunk 1 above)
Comment (troian) [reply provider/cluster/kube/inventory.go:167]: `0`. added comment to the `if`

**Comment by troian** [comment]
@hydrogen18 the provider must have multiple nodes and some of them should have CPU available less than total `CPU requests * replica count`

**Review by boz — APPROVED** [review]
:rocket:

**Review by hydrogen18 — APPROVED** [review]

**Pull request merged**